op-challenger/fault/cannon/executor_test.go

```diff
@@ -21,10 +21,9 @@ const execTestCannonPrestate = "/foo/pre.json"
 
 func TestGenerateProof(t *testing.T) {
 	input := "starting.json"
-	cfg := config.NewConfig(common.Address{0xbb}, "http://localhost:8888", config.TraceTypeCannon, true)
 	tempDir := t.TempDir()
 	dir := filepath.Join(tempDir, "gameDir")
-	cfg.CannonDatadir = tempDir
+	cfg := config.NewConfig(common.Address{0xbb}, "http://localhost:8888", config.TraceTypeCannon, true, dir)
```

Comment:
In the previous code, we were effectively setting the cannon data directory to the temp directory.

But here it looks like we now pass in the `dir` which is `filepath.Join(tempDir, "gameDir")` as opposed to just the `tempDir` value. Is this correct?

Reply:
Good spot - we should be passing `tempDir` here.  The point is to have a different directory in the config to what we pass in to the `GenerateProof` call to ensure the executor doesn't just use the value from the config. Will fix.